### PR TITLE
Implement moderated notifications

### PR DESCRIPTION
**Pull request by terceiro**
The project needs to have the "moderate notifications" flag set. With that,
every new notification will be first notified to the project admins (admin
subscriptions). The admins then are responsible for going into the admin
interface to approve the notification (there is direct link in the admin
notification email).

### CODE REVIEW

File: squad/core/tasks/__init__.py [hunk 1]
```diff
@@ -222,7 +226,20 @@ def __call__(testrun):
         testrun.status_recorded = True
         testrun.save()
 
-        ProjectStatus.create_or_update(testrun.build)
+        status = ProjectStatus.create_or_update(testrun.build)
+        if status:
+            try:
+                notify_project_status.delay(status.id)
+            except OSError as e:
```
Comment (mwasilew) [inline squad/core/tasks/__init__.py]: I got this exception in a slightly different place:

```
[2017-07-24 09:59:21 +0000] [ERROR] Task squad.core.tasks.notification.notify_project_status[b1c9e036-89f0-4e71-b5d5-5bd1af76712d] raised unexpected: ConnectionRefusedError(111, 'Connection refused')
Traceback (most recent call last):
  File "/home/milosz/linaro/tcstorm/squad_venv/lib/python3.5/site-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/home/milosz/linaro/tcstorm/squad_venv/lib/python3.5/site-packages/celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
  File "/home/milosz/linaro/tcstorm/squad/squad/core/tasks/notification.py", line 18, in notify_project_status
    send_status_notification(status)
  File "/home/milosz/linaro/tcstorm/squad/squad/core/notification.py", line 177, in send_status_notification
    notification.send()
  File "/home/milosz/linaro/tcstorm/squad/squad/core/notification.py", line 108, in send
    message.send()
  File "/home/milosz/linaro/tcstorm/squad_venv/lib/python3.5/site-packages/django/core/mail/message.py", line 348, in send
    return self.get_connection(fail_silently).send_messages([self])
  File "/home/milosz/linaro/tcstorm/squad_venv/lib/python3.5/site-packages/django/core/mail/backends/smtp.py", line 104, in send_messages
    new_conn_created = self.open()
  File "/home/milosz/linaro/tcstorm/squad_venv/lib/python3.5/site-packages/django/core/mail/backends/smtp.py", line 64, in open
    self.connection = self.connection_class(self.host, self.port, **connection_params)
  File "/usr/lib/python3.5/smtplib.py", line 251, in __init__
    (code, msg) = self.connect(host, port)
  File "/usr/lib/python3.5/smtplib.py", line 335, in connect
    self.sock = self._get_socket(host, port, self.timeout)
  File "/usr/lib/python3.5/smtplib.py", line 306, in _get_socket
    self.source_address)
  File "/usr/lib/python3.5/socket.py", line 711, in create_connection
    raise err
  File "/usr/lib/python3.5/socket.py", line 702, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [Errno 111] Connection refused
```

**Comment by mwasilew** [comment]
Overall looks OK to me. I did some manual tests on sending notifications and I found no issues. The only problem happened with attempt to send email notification without email backend. The exception happens not in the place where it's handled.

File: squad/core/tasks/__init__.py (hunk 1 above)
Comment (terceiro) [inline squad/core/tasks/__init__.py]: I assume it was in the celery worker process? I don't think there is much of a point in handling that there ...

Comment (mwasilew) [inline squad/core/tasks/__init__.py]: yes, it was in the worker.

**Pull request merged**